### DISCUSSION TITLE
How does the naming argument of Storage::disk('google')->put() work?

### DISCUSSION THREAD

**Discussion by Arturexe** [masbug/flysystem-google-drive-ext#79] [Q&A]
When I send an image object from DOM input file type element directly as http request through this extended `Storage facade`:

```
<?php

namespace App\Providers;

use Illuminate\Support\ServiceProvider;
use Illuminate\Support\Facades\Storage;
use Google\Client;
use Google\Service\Drive;
use Masbug\Flysystem\GoogleDriveAdapter;
use League\Flysystem\Filesystem;
use Illuminate\Filesystem\FilesystemAdapter;


class GoogleDriveServiceProvider extends ServiceProvider
{

    public function register()
    {
        //
    }

    public function boot()
    {
        try {
            Storage::extend('google', function($app, $config) {
                $options = [];

                if (!empty($config['teamDriveId'] ?? null)) {
                    $options['teamDriveId'] = $config['teamDriveId'];
                }

                $client = new Client();
                $client->setClientId($config['clientId']);
                $client->setClientSecret($config['clientSecret']);
                $client->refreshToken($config['refreshToken']);

                $service = new Drive($client);
                
                $adapter = new GoogleDriveAdapter($service, $config['folder'] ?? '/', $options);
                $driver = new Filesystem($adapter);

                return new FilesystemAdapter($driver, $adapter);
            });
        } catch(\Exception $e) {
            return $e->getMessage();
        }
    }
}
```

and do not specify the first argument:
```
Storage::disk('google')->put('', $request->image);
```

the image gets stored in the specified folder by `$config['folder']`. When I try to store a decoded `base64` image though, I have to specify the first argument:

```
Storage::disk('google')->put('imageName.jpg', base64_decoded($base64String));
```

which will give the image its name and place it in the same folder. If I don't specify the name argument, the image won't arrive in drive.

When I specify the name string with an image object from the DOM element:

```
Storage::disk('google')->put('imageName.jpg', $request->image);
```

drive creates a subfolder named `imageName.jpg` and puts the image inside, naming it after its `image ID`.

How does the `put()` method process its first argument? Why the different behaviour? Is this specific to [flysystem-google-drive-ext](https://github.com/masbug/flysystem-google-drive-ext)?

**Comment by erikn69** [comment]
The put method comes from laravel and phpleague/flysystem, this package is acording the flysystem testing, any question on those packages

**Comment by parallels999** [comment]
If you want to know how it works read the source code
